pkg/policy/mapstate.go

```diff
@@ -502,6 +502,38 @@ func (msm *mapStateMap) ForEachBroaderCIDRDescendant(key Key, prefixes []netip.P
 	})
 }
 
+// ForEachPortProtoAncestor iterates over broader or equal port/proto entries in the trie.
+func (msm *mapStateMap) ForEachPortProtoAncestor(key Key, f func(Key, MapStateEntry) bool) {
+	msm.trie.Ancestors(key.PrefixLength(), key, func(prefix uint, lpmKey bitlpm.Key[Key], idSet IDSet) bool {
+		k := lpmKey.Value()
+		if k.TrafficDirection == key.TrafficDirection {
```

Comment:
Traffic direction is already indexed by trie. There is no need to do this comparison.

Reply:
Fixed.